### ISSUE TITLE
Should __proto__ property be treated specially?

### ISSUE THREAD

**Issue by LongTengDao** [json5/json5#199]
The json5-spec didn't mention it.

```js
JSON5.parse('{__proto__:1}').__proto__ // not 1
```

I know it's in js-spec, I just wonder whether it should be same like js.

```js
({__proto__:1}).__proto__ // not 1
```

Because in JSON land, `__proto__` key is just a normal key, as a data language.

```js
JSON.parse('{"__proto__":1}').__proto__ // is 1
```

---

BTW: Is it right for JSON5 parser use `[[set]]` not `[[define]]`?

```js
Object.defineProperty(Object.prototype, 'xxx', { set (value) { console.log('setting!'); } });
JSON5.parse('{xxx:1}');// setting!
```

**Comment by jordanbtucker** [comment]
JSON5 should definitely maintain backward compatibility with JSON. Thank you for reporting this. I've opened PR #200 for this issue.

**Comment by jasonkhanlar** [comment]
I also am looking for a PHP implementation of JSON5

**Comment by jordanbtucker** [comment]
@jasonkhanlar [In the Wild](https://github.com/json5/json5/wiki/In-The-Wild) on the Wiki is the place to look for implementations.

**Comment by jordanbtucker** [comment]
Fixed in 755629e3485012b96cf8307e16b6b152ad2fb1e6
